modin/pandas/indexing.py

```diff
@@ -236,6 +236,18 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, item):
         row_loc, col_loc, _, __, ___ = _parse_tuple(key)
+        if isinstance(row_loc, list) and len(row_loc) == 1:
+            if row_loc[0] not in self.qc.index:
+                index = self.qc.index.insert(len(self.qc.index), row_loc[0])
+                self.qc = self.qc.reindex(labels=index, axis=0)
+                self.df._update_inplace(new_query_compiler=self.qc)
+
+        if isinstance(col_loc, list) and len(col_loc) == 1:
+            if col_loc[0] not in self.qc.columns:
+                columns = self.qc.columns.insert(len(self.qc.columns), col_loc[0])
```

Comment:
I see what you mean. so I could either insert an empty column to the DF. 
Not sure how more efficient it would be since we still use `__setitem__` at the bottom. 
```py
self.df.insert(loc=len(self.df.columns), column=col_loc[0], value=np.nan)
self.qc = self.df._query_compiler
```
but the more efficient approach would probably be to insert the new column with the values already at the correct row locations and not executing the `__setitem__` at all. 
so the function will look something like this
```py
    def __setitem__(self, key, item):
        row_loc, col_loc, _, __, ___ = _parse_tuple(key)
        if isinstance(row_loc, list) and len(row_loc) == 1:
            if row_loc[0] not in self.qc.index:
                index = self.qc.index.insert(len(self.qc.index), row_loc[0])
                self.qc = self.qc.reindex(labels=index, axis=0)
                self.df._update_inplace(new_query_compiler=self.qc)

        if isinstance(col_loc, list) and len(col_loc) == 1 and col_loc[0] not in self.qc.columns:
            new_col = pandas.Series(index=self.df.index)
            new_col[row_loc] = item
            self.df.insert(loc=len(self.df.columns), column=col_loc[0], value=new_col)
            self.qc = self.df._query_compiler
        else:
            row_lookup, col_lookup = self._compute_lookup(row_loc, col_loc)
            super(_LocIndexer, self).__setitem__(row_lookup, col_lookup, item)
```
I run several scenarios and this seems to cover it, but I could be missing something :). 
@devin-petersohn can you please take a look before I push this

Reply:
Thanks @eavidan, that looks great. I think `insert` without doing a `__setitem__` is the right approach.